src/autopas/containers/verletClusterLists/VerletClusterListsRebuilder.h

```diff
@@ -86,8 +86,8 @@ class VerletClusterListsRebuilder {
     const auto numTowersNew = numTowersPerDim[0] * numTowersPerDim[1];
 
     // collect all particles that are now not in the right tower anymore
-    auto invalidParticles = collectOutOfBoundsParticlesFromTowers();
-    // collect all remaining particles that are not yet assigned to towers
+    std::vector<std::vector<Particle>> invalidParticles;
+    //  collect all particles that are not yet assigned to towers
     invalidParticles.push_back(std::move(_particlesToAdd));
```

Comment:
```suggestion
    std::vector<std::vector<Particle>> invalidParticles{};
    // Reserve for:
    //   - _particlesToAdd (+1)
    //   - surplus towers (+max(0, numTowersNew - numTowersOld))
    //   - particles out of bounds of new towers (+numTowersNew)
    invalidParticles.reserve(1 + std::max(0, static_cast<int>(numTowersOld) - static_cast<int>(numTowersNew)) + numTowersNew)
    // collect all particles that are not yet assigned to towers
    invalidParticles.push_back(std::move(_particlesToAdd));
```

- [x] formatting
- [x] reserve before `push_back` / `insert`
- [x] casts before substracting unsigned integers

...should be right I think?